src/Reduce.php

```diff
@@ -175,4 +175,55 @@ public static function isReversed(iterable $data): bool
 
         return true;
     }
+
+    /**
+     * Returns true if all given collections are the same.
+     *
+     * For single iterable or empty iterables list returns true.
+     *
+     * @param iterable<mixed> ...$iterables
+     *
+     * @return bool
+     */
+    public static function same(iterable ...$iterables): bool
+    {
+        try {
+            foreach (Multi::zipEqual(...$iterables) as $values) {
+                foreach (Single::pairwise($values) as [$lhs, $rhs]) {
+                    if ($lhs !== $rhs) {
+                        return false;
+                    }
+                }
+            }
+        } catch (\LengthException $e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if all given collections have the same lengths.
+     *
+     * For single iterable or empty iterables list returns true.
+     *
+     * @param iterable<mixed> ...$iterables
+     *
+     * @return bool
+     */
+    public static function sameCount(iterable ...$iterables): bool
+    {
+        $count = null;
+        foreach ($iterables as $iterable) {
+            $currentCount = self::toCount($iterable);
+
+            if ($count !== null && $count !== $currentCount) {
+                return false;
+            }
+
+            $count = $currentCount;
+        }
+
+        return true;
```

Comment:
I wonder if this could be simplified by using some functional programming constructs.

For example, would something like this work and be easier to follow:
```php
if (count($iterables) <= 1) {
    return true;
}

$counts = array_map('self::toCount', $iterables);
return count(array_unique($counts)) === 1;
```

Reply:
Done.